torch/_inductor/kernel/flex_decoding.py

```diff
@@ -426,14 +426,14 @@ def create_flex_decoding_kernel(*args, **kwargs):
             BLOCK_M=BLOCK_M,
             SPLIT_KV=SPLIT_KV,
             BLOCK_DMODEL=query.get_size()[-1],
-            SM_SCALE=scale,
+            SM_SCALE=kernel_params["scale"],
             # Performance tuning
             BLOCK_N=BLOCK_N,
             # For now, we always assume the "sound" option
-            ROWS_GUARANTEED_SAFE=False,
+            ROWS_GUARANTEED_SAFE=kernel_params["rows_guaranteed_safe"],
```

Comment:
I would like to just pass it in as an unpacked dict kwarg here. In particular, I would like to also be able to use kernel options to set `BLOCK_N` and `BLOCK_M`.